pkg/scheduler/framework/v1alpha1/registry_test.go

```diff
@@ -104,3 +104,163 @@ func loadConfig(data []byte) (*config.KubeSchedulerConfiguration, error) {
+			},
+			expected: Registry{
+				"pluginFactory1": NewMockNoopPluginFactory(),
+			},
+			shouldError: true,
+		},
+	}
+
+	for scenarioName, scenario := range tests {
+		err := scenario.primaryRegistry.Merge(scenario.registryToMerge)
+
+		if err == nil && scenario.shouldError == true {
+			t.Errorf("Test %s: Merge() Failed to return Error.", scenarioName)
+			continue
+		}
+
+		if !isRegistryEqual(scenario.expected, scenario.primaryRegistry) {
+			t.Errorf("Test %s: Failed to Merge(). Expected %v. Got %v instead.", scenarioName, scenario.expected, scenario.primaryRegistry)
+		}
+	}
+}
+
+func TestRegister(t *testing.T) {
+	tests := map[string]struct {
+		registry Registry
+		nameToRegister string
+		factoryToRegister PluginFactory
+		expected Registry
+		shouldError bool
+	}{
+		"goodRegister": {
+			registry: Registry{},
+			nameToRegister: "pluginFactory1",
+			factoryToRegister: NewMockNoopPluginFactory(),
+			expected: Registry{
+				"pluginFactory1": NewMockNoopPluginFactory(),
+			},
+			shouldError: false,
+		},
+		"badRegisterDuplicatePluginFactory": {
+			registry: Registry{
+				"pluginFactory1": NewMockNoopPluginFactory(),
+			},
+			nameToRegister: "pluginFactory1",
+			factoryToRegister: NewMockNoopPluginFactory(),
+			expected: Registry{
+				"pluginFactory1": NewMockNoopPluginFactory(),
+			},
+			shouldError: true,
+		},
+	}
+
+	for scenarioName, scenario := range tests {
+		err := scenario.registry.Register(scenario.nameToRegister, scenario.factoryToRegister)
+
+		if err == nil && scenario.shouldError == true {
+			t.Errorf("Test %s: Register() Failed to return Error.", scenarioName)
+			continue
+		}
+
+		if !isRegistryEqual(scenario.expected, scenario.registry) {
+			t.Errorf("Test %s: Failed to Register(). Expected %v. Got %v instead.", scenarioName, scenario.expected, scenario.registry)
+		}
+	}
+}
+
+func TestUnregister(t *testing.T) {
+	tests := map[string]struct {
+		registry Registry
+		nameToUnregister string
+		expected Registry
+		shouldError bool
+	}{
+		"goodUnregister": {
+			registry: Registry{
+				"pluginFactory1": NewMockNoopPluginFactory(),
+				"pluginFactory2": NewMockNoopPluginFactory(),
+			},
+			nameToUnregister: "pluginFactory1",
+			expected: Registry{
+				"pluginFactory2": NewMockNoopPluginFactory(),
+			},
+			shouldError: false,
+		},
+		"badUnregisterMissingPluginFactory": {
+			registry: Registry{},
+			nameToUnregister: "pluginFactory1",
+			expected: Registry{},
+			shouldError: true,
+		},
+	}
+
+	for scenarioName, scenario := range tests {
+		err := scenario.registry.Unregister(scenario.nameToUnregister)
+
+		if err == nil && scenario.shouldError == true {
+			t.Errorf("Test %s: Unregister() Failed to return Error.", scenarioName)
+			continue
+		}
+
+		if !isRegistryEqual(scenario.expected, scenario.registry) {
+			t.Errorf("Test %s: Failed to Register(). Expected %v. Got %v instead.", scenarioName, scenario.expected, scenario.registry)
+		}
+	}
+}
```

Comment:
missing new line.

Reply:
@neolit123 Is there an issue with the pull-kubernetes-verify check? It is telling me that the gofmt command thinks there should be no new line:
```
diff -u ./pkg/scheduler/framework/v1alpha1/registry_test.go.orig ./pkg/scheduler/framework/v1alpha1/registry_test.go
--- ./pkg/scheduler/framework/v1alpha1/registry_test.go.orig	2020-01-02 01:19:28.691823500 +0000
+++ ./pkg/scheduler/framework/v1alpha1/registry_test.go	2020-01-02 01:19:28.691823500 +0000
@@ -263,4 +263,3 @@
 		}
 	}
 }
-

Run ./hack/update-gofmt.sh
```

Reply:
are you not using line-feed LF in your editor for the new line?
otherwise, i don't have guesses why this might happen.

files should always end up with a new line.